### DISCUSSION TITLE
Using Python with gitoxide

### DISCUSSION THREAD

**Discussion by damascene** [GitoxideLabs/gitoxide#1279] [Q&A] [answered]
### Summary 💡

I would like to thank you for your contributions to important open source software. I wanted to use GitPython but I noticed a message directing here. However I don't see instructions on using python with gitoxide. I thought there might be a wrapper that allows to call the Rust tool from Python.  

Is there a plan for any?

**Comment by Byron** [comment, marked as answer]
Thanks a lot for your interest! 

Unfortunately `gitoxide` isn't yet available for use in Python, one will have to use Rust to do that. I have added a note to the text that let you here to make that clear as well.

It will probably be a while until that changes, as `gitoxide` will first have to reach feature parity with `git2`, be stabilised with a version 1.0, and then finally can spread with bindings to other programming languages.

**Comment by velle** [comment]
> It will probably be a while until that changes

I know it's probably not simple to answer, but do you have any idea if this is in 1 year, 3 years or 15 years?

I really appreciate the "meta" communication on GitPython, namely [A peek into the future](https://github.com/gitpython-developers/GitPython?tab=readme-ov-file#gitoxide-a-peek-into-the-future) and [Development Status](https://github.com/gitpython-developers/GitPython?tab=readme-ov-file#development-status). 

It could be _even_ better if you added the approximate time at which you started developing Gitoxide and the time you decided that GitPython went into maintainance mode. And, if possible, any kind of prognosis on when a Python binding for Gitoxide can be expected. 

Thanks for all the great work :)

**Reply by Byron** [reply]
Knowing all predictions further than 2 hours ahead are very wrong, I'd hope that in 3-5 years there will be bindings that aren't maintained by me. This means `gitoxide` is stable, en-par with `git2` in terms of features, and wide-spread enough to make users of `pygit2` want to use `pygix` instead.
Plenty of conditions have to hold to make this happen, but for now I still looks like `gitoxide` is on-track.
